integration_test_suite/integration_tests/src/test/java/io/prometheus/jmx/test/http/ssl/SSLWithTrustStoreAndClientAuth.java

```diff
@@ -0,0 +1,277 @@
+        // format or don't support the required TLS cipher suites
+        return ExporterTestEnvironment.createExporterTestEnvironments()
+                .filter(new PKCS12KeyStoreExporterTestEnvironmentFilter())
+                .map(exporterTestEnvironment -> exporterTestEnvironment.setBaseUrl(BASE_URL));
+    }
+
+    @Verifyica.Prepare
+    public static void prepare(ClassContext classContext) {
+        TestSupport.getOrCreateNetwork(classContext);
+    }
+
+    @Verifyica.BeforeAll
+    public void beforeAll(ArgumentContext argumentContext) {
+        Class<?> testClass = argumentContext.classContext().testClass();
+        Network network = TestSupport.getOrCreateNetwork(argumentContext);
+        TestSupport.initializeExporterTestEnvironment(argumentContext, network, testClass);
+    }
+
+    private SSLContext initSSLContextForClientAuth(JmxExporterMode mode) {
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+
+            // to verify cert auth with existing test pki resources, use self-signed server cert as
+            // client cert and source of trust
+            final String type = "PKCS12";
+            final char[] password = "changeit".toCharArray();
+            final String keyStoreResource =
+                    Strings.formatIfArgs(
+                            "%s/%s/localhost.pkcs12",
+                            this.getClass().getSimpleName(), mode.toString());
+            KeyStore keyStore = KeyStore.getInstance(type);
+            try (InputStream inputStream = this.getClass().getResourceAsStream(keyStoreResource)) {
+                keyStore.load(inputStream, password);
+            }
+            KeyManagerFactory km =
+                    KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            km.init(keyStore, password);
+            TrustManagerFactory tm =
+                    TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tm.init(keyStore);
+
+            sslContext.init(
+                    km.getKeyManagers(), tm.getTrustManagers(), new java.security.SecureRandom());
+
+            return sslContext;
+        } catch (IOException
+                | NoSuchAlgorithmException
+                | KeyStoreException
+                | CertificateException
+                | KeyManagementException
+                | UnrecoverableKeyException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Verifyica.Test
+    @Verifyica.Order(1)
+    public void testHealthy(ExporterTestEnvironment exporterTestEnvironment) throws IOException {
+
+        String url = exporterTestEnvironment.getUrl(ExporterPath.HEALTHY);
+
+        try {
+            HttpResponse httpResponse = HttpClient.sendRequest(url);
+            throw new AssertionFailedError("expected exception on no client cert");
+        } catch (Exception expectedOnNoClientCert) {
+        }
```

Comment:
Change this to use Assertj `assertThatExceptionOfType`

Reply:
Thanks, done. much nicer.